### ISSUE TITLE
Switch to C++20

### ISSUE THREAD

**Issue by jovany-wang** [purecpp-org/raftcpp#210]

**Comment by jovany-wang** [comment]
@qicosmos Has restrpc supported C++20 yet?

**Comment by qicosmos** [comment]
I want to use a solid coroutine library to improve rest_rpc, i think the coroutine library is going to publish soon.

**Comment by jovany-wang** [comment]
@devillove084 CC Could you help to investigate that when it's ready for switching to C++20. I believe these should be done before:
- [ ] Switching to grpc
- [ ] Investigate if all our dependencies are support C++20

**Comment by jovany-wang** [comment]
It's already done.
